pkg/dart2native/lib/generate.dart

```diff
@@ -182,6 +191,13 @@ class _Generator {
         _sourcePath = _normalize(sourceFile)!,
         _packages = _normalize(packages) {
     if (_kind == Kind.exe) {
+      if (_targetArch == null) {
+        throw ArgumentError('targetArch must be specified for executables.');
```

Comment:
I believe the Flutter team once went over the errors the tooling could throw and added clear explanation points on how to move forward. I would expect such practice to also be present here.

If I encounter "targetArch must be specified for executables" I would ask myself:
- "what is targetArch", is an abbreviation for "target architecture" (see [Effective Dart for abbreviations](https://dart.dev/effective-dart/documentation#avoid-abbreviations-and-acronyms-unless-they-are-obvious))
- How can I specify so? Is it a parameter flag `--targetArch`>
- What are the available architectures I can compile into? Is there documentation stating all the supported architectures? Maybe we can point the reader to: https://github.com/dart-lang/sdk/blob/main/docs/Supported-Architectures.md?
- What architecture is my system running? Could the message include a suggested resolution with the current architecture my system is using?

Reply:
> "what is targetArch", is it an abbreviation for "target architecture"?

Yes, it is an abbreviation for "target architecture". I was thinking `targetCPU` but in my experience, that usually is for specific CPU optimizations (eg. zen1, skylake, generic, baseline).

> How can I specify so? Is it a parameter option `--targetArch`? Is there an example on how to specify it?

`--target-arch` was added to pair with `--target-os`. However, `--target-os` doesn't support cross compiling and so I haven't been able to get `--target-arch` to cross compile. I at least added the logic for `--target-arch` to pair with the existing `--target-os` flag.

> What are the available architectures I can compile into? Is there documentation stating all the supported architectures? Maybe we can point the reader to: https://github.com/dart-lang/sdk/blob/main/docs/Supported-Architectures.md? Note that these are already in the "help command with TargetArch.names, but it is not as clear as the table provided in the documentation.

The `TargetArch.names` I added in `pkg/vm/lib/target_arch.dart` was based on that table + observations I saw when using `grep` to figure out how this repo works.

> What architecture is my system running? Could the message include a suggested resolution with the current architecture my system is using?

Could you provide more information to what this sort of means?

> For example, a new improved message would be (ignore bad formatting and bad command

Should we also change the one for the target OS?